### PR TITLE
Apriltag

### PR DESCRIPTION
**Pull request by Hwang1907**
i am trying to put my code in apriltag branch to main branch

### CODE REVIEW

File: build.gradle
Comment (Gold856) [inline build.gradle:1]: Sync to main

File: src/main/java/frc/robot/SwerveModule.java [hunk 1]
```diff
@@ -21,7 +21,7 @@
 /**
  * Contains all the hardware and controllers for a swerve module.
  */
-public class SwerveModule {
+public class SwerveModule implements AutoCloseable {
```
Comment (Gold856) [inline src/main/java/frc/robot/SwerveModule.java:24]: Stuff doesn't need closing...

File: src/main/java/frc/robot/subsystems/DriveSubsystem.java
Comment (Gold856) [inline src/main/java/frc/robot/subsystems/DriveSubsystem.java:1]: Doesn't need closing, remove singleton code

**Review by Gold856 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/SwerveModule.java (hunk 1 above)
Comment (jhhbrown1) [reply src/main/java/frc/robot/SwerveModule.java:24]: 1. Doesn't seem to hurt to close SwerveModules.
2. Probably better to do that if you want to run a variety of unit tests. See: https://docs.wpilib.org/en/stable/docs/software/wpilib-tools/robot-simulation/unit-testing.html
3. No SwerveModules using the ports (e.g., in other unit tests) can be instantiated until you close them.

File: src/main/java/frc/robot/subsystems/DriveSubsystem.java
Comment (jhhbrown1) [reply src/main/java/frc/robot/subsystems/DriveSubsystem.java:1]: There is a very strong reason for the use of the singleton pattern in WPILib programming and it was the norm in previous years. Also, if we choose not to use it this time, some constructors of commands/subsysems may need a longer list of parameters. Why not using it this time? What's the benefit?

**Pull request closed without merging**